chain/client/src/client.rs

```diff
@@ -2374,12 +2374,11 @@ impl Client {
                 state_sync_info.shards.iter().map(|tuple| tuple.0).collect();
             // Notify each shard to sync.
             if notify_state_sync {
```

Comment:
Please have a look at how notify_state_sync works. It's a bit crazy but it's actually reused across the different state syncs and once one will set it to true all the following ones will also have it set. I would move that variable to inside of the for loop. 

I discussed it briefly with @VanBarbascu and he may have an even better place to put the unloading so stay tuned for that too.